numpy/ma/core.py

```diff
@@ -3184,13 +3180,17 @@ def __setitem__(self, indx, value):
 
         """
         # 2016.01.15 -- v1.11.0
-        warnings.warn(
-           "Currently, slicing will try to return a view of the data," +
-           " but will return a copy of the mask. In the future, it will try" +
-           " to return both as views. This means that using `__setitem__`" +
-           " will propagate values back through all masks that are present.",
-           FutureWarning
-        )
+        self._oldsharedmask = getattr(self, "_oldsharedmask", False)
+        self._oldsharedmask = self._oldsharedmask or self._sharedmask
```

Comment:
I guess the idea here is that `_oldsharedmask` propagates, but `_sharedmask` doesn't for now.